pkg/telemetry/phonehome/request_params.go

```diff
@@ -48,3 +49,12 @@ func (rp *RequestParams) HasPathIn(patterns []string) bool {
 func (rp *RequestParams) Is(s *ServiceMethod) bool {
 	return rp.Method == s.GRPCMethod || (rp.Method == s.HTTPMethod && rp.PathMatches(s.HTTPPath))
 }
+
+// GetGRPCRequestBody sets the output body argument. Returns io.EOF on empty body.
+func GetGRPCRequestBody[T any](rp *RequestParams, body **T) error {
```

Comment:
Why do you use a double pointer construction here instead of returning a pointer to the body via `(T*, error)` return type?

Reply:
Go cannot infer the type by return value, unfortunately, and I don't have a way to get the right type for explicit type specification.